src/interop.hpp

```diff
@@ -400,6 +402,28 @@ namespace openre::interop
     bool operator==(const save_state& lhs, const save_state& rhs);
     bool operator!=(const save_state& lhs, const save_state& rhs);
 
+    class memory_comparer
+    {
+    private:
+        uintptr_t _begin = 0;
+        uintptr_t _end = 0;
+        save_state _pre;
+        save_state _first;
+        save_state _second;
```

Comment:
Should this `save_state` be a class member if it's only used in the `log `function for logging purposes?